src/mscorlib/shared/System/StringSpanHelpers.cs

```diff
@@ -149,8 +151,38 @@ public static void CheckStringComparison(StringComparison comparisonType)
             // Single comparison to check if comparisonType is within [CurrentCulture .. OrdinalIgnoreCase]
             if ((uint)(comparisonType - StringComparison.CurrentCulture) > (StringComparison.OrdinalIgnoreCase - StringComparison.CurrentCulture))
             {
-                throw new ArgumentException(SR.NotSupported_StringComparison, nameof(comparisonType));
+                ThrowStringComparison_NotSupported();
             }
         }
+
+        public static CompareOptions GetCaseCompareOfComparisonCulture(StringComparison comparisonType)
+        {
+            Debug.Assert((uint)(comparisonType - StringComparison.CurrentCulture) <= (StringComparison.InvariantCultureIgnoreCase - StringComparison.CurrentCulture));
+
+            // Culture enums can be & with CompareOptions.IgnoreCase 0x01 to extract if IgnoreCase or CompareOptions.None 0x00
+            //
+            // CompareOptions.None                          0x00
+            // CompareOptions.IgnoreCase                    0x01
+            //
+            // StringComparison.CurrentCulture:             0x00
+            // StringComparison.InvariantCulture:           0x02
+            //
+            // StringComparison.CurrentCultureIgnoreCase:   0x01
+            // StringComparison.InvariantCultureIgnoreCase: 0x03
+
+            return (CompareOptions)((int)comparisonType & (int)CompareOptions.IgnoreCase);
+        }
+
+        [StackTraceHidden]
+        private static void ThrowStringComparison_NotSupported()
+        {
+            throw GetStringComparisonException_NotSupported();
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
Doesn't JIT automatically not-inline methods that unconditionally throw? (`InlineObservation::CALLEE_DOES_NOT_RETURN`)

Reply:
This method doesn't throw; it just constructs and returns the exception to the method that throws.

Issue is if the throw method does too much then `basicBlockCount` can be greater than `1` then it no longer counts as a non-returning throw method. Moving the exception constructor, the resource lookup and marking that as non-inlining; and then just throwing the return guarantees that the `basicBlockCount` is always `1` in the throw method. `NoInlining` probably isn't needed here; its a trait for older runtimes that don't know about non-returning methods.

Reply:
> This method doesn't throw; it just constructs and returns the exception to the method that throws.

am I misreading? I see `throw new ArgumentException(...`

Reply:
eugh; should be `return ` rather than `throw`

Reply:
Can we add to and use ThrowHelper here instead (https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/ThrowHelper.cs)?